ranking/src/contest_config.rs

```diff
@@ -0,0 +1,56 @@
+use super::read_codeforces::{fetch_cf_contest};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+/// Represents the outcome of a contest
+#[derive(Serialize, Deserialize)]
+pub struct Contest {
+    pub id: usize,
+    pub name: String,
+    pub time_seconds: usize,
+    pub standings: Vec<(String, usize, usize)>,
+}
+
+pub enum ContestSource {
+	Codeforces,
+	Reddit,
+	StackOverflow
+}
+
+pub struct ContestConfig {
+	pub contest_id_file : String,
+	pub contest_cache_folder : String,
+}
+
+pub fn get_contest_config(source: ContestSource) -> ContestConfig {
+	let source_name = match source {
+		ContestSource::Codeforces => "codeforces",
+		ContestSource::Reddit => "reddit",
+		ContestSource::StackOverflow => "stackoverflow"
+	};
+
+	let config = ContestConfig {
+		contest_id_file: String::from(format!("../data/{}/contest_ids.json", source_name)),
+		contest_cache_folder: String::from(format!("../cache/{}", source_name)),
+	};
+	config
+}
+
+/// Get a list of all the contest IDs in chronological order
+pub fn get_contest_ids(contest_id_file: &String) -> Vec<usize> {

```

Comment:
`&str` is usually a better (more general) argument type than `&String`, but I think the best option is to make it generic in `AsRef<Path>`, just like what we have in `fetch_cf_contest()`.

In case it helps, `String` is to `str` as `Vec<T>` is to `[T]` as `PathBuf` is to `Path`. The first in each pair has a pointer to an owned allocation on the heap, whereas the second *is* the allocation.